WordPress/Classes/Services/SharingSyncService.swift

```diff
@@ -0,0 +1,111 @@
+            return nil
+        }
+
+        return SharingServiceRemote(wordPressComRestApi: api)
+    }
+
+    /// Syncs Publicize connections for the specified wpcom blog.
+    ///
+    /// - Parameters:
+    ///     - blog: The `Blog` for which to sync publicize connections
+    ///     - success: An optional success block accepting no parameters.
+    ///     - failure: An optional failure block accepting an `NSError` parameter.
+    ///
+    @objc open func syncPublicizeConnectionsForBlog(_ blog: Blog, success: (() -> Void)?, failure: ((NSError?) -> Void)?) {
+        let blogObjectID = blog.objectID
+        guard let remote = remoteForBlog(blog) else {
+            failure?(SharingSyncServiceError.siteWithNoRemote as NSError)
+            return
+        }
+
+        remote.getPublicizeConnections(blog.dotComID!, success: { remoteConnections in
+
+            // Process the results
+            self.mergePublicizeConnectionsForBlog(blogObjectID, remoteConnections: remoteConnections, onComplete: success)
+        },
+        failure: failure)
+    }
+
+    /// Called when syncing Publicize connections. Merges synced and cached data, removing
+    /// anything that does not exist on the server.  Saves the context.
+    ///
+    /// - Parameters:
+    ///     - blogObjectID: the NSManagedObjectID of a `Blog`
+    ///     - remoteConnections: An array of `RemotePublicizeConnection` objects to merge.
+    ///     - onComplete: An optional callback block to be performed when core data has saved the changes.
+    ///
+    fileprivate func mergePublicizeConnectionsForBlog(_ blogObjectID: NSManagedObjectID, remoteConnections: [RemotePublicizeConnection], onComplete: (() -> Void)?) {
+        coreDataStack.performAndSave({ context in
+            var blog: Blog
+            do {
+                blog = try context.existingObject(with: blogObjectID) as! Blog
+            } catch let error as NSError {
+                DDLogError("Error fetching Blog: \(error)")
+                // Because of the error we'll bail early, but we still need to call
+                // the success callback if one was passed.
+                return
+            }
+
+            let currentPublicizeConnections = self.allPublicizeConnections(for: blog, in: context)
+
+            // Create or update based on the contents synced.
+            let connectionsToKeep = remoteConnections.map { (remoteConnection) -> PublicizeConnection in
+                let pubConnection = PublicizeConnection.createOrReplace(from: remoteConnection, in: context)
+                pubConnection.blog = blog
+                return pubConnection
+            }
+
+            // Delete any cached PublicizeServices that were not synced.
+            for pubConnection in currentPublicizeConnections {
+                if !connectionsToKeep.contains(pubConnection) {
+                    context.delete(pubConnection)
+                }
+            }
+        }, completion: { onComplete?() }, on: .main)
```

Comment:
Should the queue here become injectable, like done in various recent PRs?

Reply:
I think that would be nice. But for now, I kept the existing code since there are too many call sites to change.